src/main/java/io/cryostat/recordings/RecordingMetadataManager.java

```diff
@@ -740,37 +783,40 @@ private boolean isTargetReachable(ConnectionDescriptor cd) {
     private boolean targetRecordingExists(ConnectionDescriptor cd, String recordingName) {
         CompletableFuture<Boolean> future = new CompletableFuture<>();
         try {
-            this.targetConnectionManager.executeConnectedTask(
-                    cd,
-                    conn -> {
-                        try {
-                            return conn.getService().getAvailableRecordings().stream()
-                                    .anyMatch(
-                                            r ->
-                                                    future.complete(
-                                                            Objects.equals(
-                                                                    recordingName, r.getName())));
-                        } catch (ConnectionException e) {
-                            if (e.getCause() instanceof SecurityException) {
-                                // don't have credentials to access target
-                                if (cd.getCredentials().isEmpty()) {
-                                    logger.warn(
-                                            "Target {} requires credentials to access recordings",
-                                            cd.getTargetId());
-                                    throw e;
-                                } else {
-                                    logger.warn(
-                                            "Target {} credentials are invalid", cd.getTargetId());
-                                    throw e;
+            return this.targetConnectionManager
+                    .executeConnectedTaskAsync(
+                            cd,
+                            conn -> {
+                                try {
+                                    return conn.getService().getAvailableRecordings().stream()
+                                            .anyMatch(
+                                                    r ->
+                                                            future.complete(
+                                                                    Objects.equals(
+                                                                            recordingName,
+                                                                            r.getName())));
+                                } catch (ConnectionException e) {
+                                    if (e.getCause() instanceof SecurityException) {
```

Comment:
These Exception specific logs should probably be removed if they were removed in the other instances as well.